File: apps/src/chat/frontend/js/packets/websocket/UserEnterRoomPacket.js
export default class UserEnterRoomPacket {

    #username;

    constructor(
        username
    ){
        this.#username = username;
    }

    toData(){
        return {
            target_username: this.#username,
            type: "user_enter_room"
        }
    }

    static fromData(data){
        return new UserEnterRoomPacket(data["username"])
    }

    getUsername(){
        return this.#username;
    }
}

File: apps/src/chat/backend/packets/websocket/user_enter_room_packet.py
from ..packet import Packet

class UserEnterRoomPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def to_data(self):
        return {
            "username": self._username,
            "type": "user_enter_room"
        }

    @staticmethod
    def from_data(data):
        return UserEnterRoomPacket(
            data["username"]
        )